### DISCUSSION TITLE
Accessing S3 remotes via Podman container?

### DISCUSSION THREAD

**Discussion by seannymurrs** [containers/podman#24217] [Q&A]
I have an S3-compatible remote mounted locally using rclone mount. I'm trying to run a container of Audiobookshelf via Podman, but the container is unable to access my locally mounted directory. I did some searching, and I've found a few posts mentioning compatibility issues between S3 and Podman, but I can't find specifics on whether there are ways to workaround this issue.

**Comment by rhatdan** [comment]
What error are you seeing?  Could be SELinux or User Namespace blocking access.

Is the HomeDir local?

**Comment by eriksjolund** [comment]
I did some experimenting with FUSE and squashfs and noticed that

| arguments to __podman run__ | result | output |
| ---                                      | --------- |  --------- |
| --user 0:0  |  success | `total 0` |
| --user=1000:1000 |  failure | `ls: cannot access '/ctr/mntdir/ubuntu': Permission denied` |
 --user=1000:1000 --userns=keep-id:uid=1000,gid=1000 | success | `total 0` |


All the details

<details>
  <summary>Click me</summary>


1. `mkdir ~/squashctr`
2. `cd ~/squashctr`
3. Create file _~/squashctr/Containerfile_ containing
   ```
   FROM ubuntu
   RUN apt-get update ; apt-get install -y squashfs-tools
   RUN mksquashfs /home /ubuntu.squash
   ```
3. `podman build -t squash .`
4. `podman run --rm --name tmpctr --detach localhost/squash sleep 200`
5. `podman cp tmpctr:/ubuntu.squash .`
6. `mkdir mntdir`
7. ` squashfuse ubuntu.squash mntdir`
8. `id -u`
   The following output is printed
   ```
   1210
   ```
9. `id -g`
   The following output is printed
   ```
   1210
   ```
10. Run command
     ```
     podman run --rm --security-opt label=disable \
            --user=0:0 \
            -v /home/test711/squashctr/:/ctr ubuntu ls -l 
    ```
    The following output is printed
    ```
    total 0
    ```
10. Run command
     ```
      podman run --rm --security-opt label=disable \
             --user=1000:1000 \
             -v /home/test711/squashctr/:/ctr ubuntu ls -l 
    ```
    The following output is printed
    ```
    ls: cannot access '/ctr/mntdir/ubuntu': Permission denied
    ```
10. Run command
    ```
    podman run --rm --security-opt label=disable \
            --user=1000:1000 --userns=keep-id:uid=1000,gid=1000  \:
            -v /home/test711/squashctr/:/ctr ubuntu ls -l 
    ```
    The following output is printed
    ```
    total 0
    ```

</details>

I somehow get the impression that `--user $uid:$gid` needs `--userns keep-id:uid=${uid},${gid}` when $uid and $gid are positive.
(I haven't fully understood how all this works)

@seannymurrs Maybe you need to add `--userns keep-id:uid=${uid},${gid}`?
